### ISSUE TITLE
Content-based filtering

### ISSUE THREAD

**Issue by WanderingStar** [Ranchero-Software/NetNewsWire#4406]
I'd like to be able to filter out (hide, mark read, or just don't download) certain articles based on what's in them.

For example, I'm subscribed to the ArsTechnica Science feed, but I am uninterested in rocketry, so would like to skip all articles that contain:

```
<category><![CDATA[space]]></category>
```

Similarly, I'd like to not see entries in the 404 Media feed about their podcast, so would like to skip articles with:

```
<category><![CDATA[Podcast]]></category>
```

I suspect the answer is "this is too far outside of NetNewsWire's scope, use a filter on the feed side", but I thought I'd ask.

**Comment by beryllium** [comment]
I would like to have keyword filtering to either A) hide news items that I'm not in the mood for (i.e., current political turmoil), or B) shuffle that content off into a separate section that I can browse when I need to spike my blood pressure.

**Comment by bbsixzz** [comment]
Same problem here currently using http://feedrinse.com/index-old.php but it's not really that great with granularity.

**Comment by gbarr** [comment]
I would like to have filtering by author. I have feeds from sites with multiple authors and some I am just not interest in. It would be great to have filters applied at download time to hide or mark as read those authors

**Comment by mathieu-gervois** [comment]
+ 1, I would also like to have filtering options. 🙏

**Comment by renato-umeton** [comment]
@brentsimmons is this implemented or in the road map? I'd love to filter for `<category><![CDATA[AI]]></category>` and similar. Thank you very much for considering.

**Comment by renato-umeton** [comment]
@brentsimmons - there is and implementation suggested in https://github.com/Ranchero-Software/NetNewsWire/pull/5216
Thank you very much for considering.

**Comment by johniannone** [comment]
+1 here, and to have parts of the URL as a filtering option, such as for YouTube:

* .../shorts ❌
* .../watch ✅

**Comment by spacecowgoesmoo** [comment]
I'd also love to have this as a tool to fight enshittification. A primary concern for feeds I subscribe to is: how much meaningless engagement bait, and ads presented as news do they include? It's essentially adblock. Here's some breaking news from MacRumors:

<img width="403" height="89" alt="Image" src="https://github.com/user-attachments/assets/3ee3181b-94dc-4f43-9ebd-99a6711fa427" />

**Comment by fortinmike** [comment]
My 2¢:

Filtering is essential because some (many) feeds have a lower SNR than others but are still worthwhile. @spacecowgoesmoo has mentioned fighting enshittification as a reason, and I strongly agree. For me, this has been a pain point of RSS for _years_, and subscribing to a paid service just for filtering makes no financial sense when everything could clearly be done at the edge for free.

I've implemented my own self-hosted feed filtering server named [feed-sieve](https://github.com/fortinmike/feed-sieve) and that has been a huge personal relief and has been serving me fine -- feel free to take a look and give me feedback -- but I would be very happy to retire it in favour of something built right into NetNewsWire, as using that server (which acts as a kind of proxy for feeds) has its compromises.

Here's what the rule definition UI looks like, for reference (with two good filtering examples BTW, those articles are so pointless and numerous during beta and WWDC periods):

<img width="722" height="397" alt="Image" src="https://github.com/user-attachments/assets/1488c04d-be70-4ca5-a521-6844922d8dcc" />

I've been contemplating contributing a similar filtering feature to NetNewsWire eventually but was not sure about the willingness of NNW developers to accept such a feature (and it might be best for people familiar with the codebase to do the actual implementation).

I've also thought about using local AI to intelligently filter (as some suggested) but also deduplicate news across feeds. Say 10 of your feeds are relaying almost the same news, it would make sense to keep only one high-quality article to reduce the noise. Or maybe optionally generate a synthetic article to replace them all. But didn't get to implementing that yet.

Another nice feature of feed-sieve that is not strictly filtering is (currently hardcoded) auto-tagging of feed items based on content. The one hardcoded case here is automatic prefixing of YouTube shorts item titles with "[Short]", which enables at-a-glance to see which items are quick watches and which are full-length videos. That allows me to binge-watch a few shorts and reduce my unread count quickly instead of items piling up. Or filter out shorts entirely.

In short, I have lots of ideas and opinions on what good filtering and filtering-adjacent features might look like. I'd like to be part of the discussion on making this a reality in NetNewsWire (and make my own self-hosted feed proxy irrelevant in the process)! 🙂

**Comment by bbsixzz** [comment]
What a great coincidence then:

Time to Start Doing Feature Requests
https://inessential.com/2026/08/26/after-fixin-the-bugs.html

**Comment by fortinmike** [comment]
Not really a coincidence, that post and seeing content filtering mentioned there is why I decided to chime in 😄

But yeah, the perspective of NetNewsWire gaining some long sought-after features is appealing.
